docs/docs/resources/pipeline.yaml

```diff
@@ -0,0 +1,334 @@
+# `pipeline.yaml` contains the configurations for all components to be deployed
+#
+# In order to avoid repetition (for example with `namespace`), the user can set
+# values in `defaults.yaml` for all apps of a certain type (could be specific
+# like `producer` or broad like `kubernetes-app`). ]
+
+# Base Kubernetes App
+- type: kubernetes-app
+  name: kubernetes-app # required
+  namespace: namespace # required
+  # `app` contains application-specific settings, hence it does not have a rigid
+  # structure. The fields below are just an example.
+  app: # required
+    key1:
+      key4: value3
+    key2: value1
+    key3: value2
+  # Topic(s) from which the component will read input
+  from:
+    topics: # required
+      ${pipeline_name}-input-topic:
+        type: input # required
+        role: topic-role
+  # Topic(s) into which the component will write output
+  to:
+    topics: # required
+      ${pipeline_name}-output-topic:
+        type: output # required
+        role: topic-role
+        keySchema: key-schema
```

Comment:
Please refer to SchemaProvider that needs to be implemented

Reply:
done